dbfit-java/sqlserver/src/main/java/dbfit/environment/SqlServerEnvironment.java

```diff
@@ -80,15 +78,18 @@ protected String parseCommandText(String commandText) {
             String query) throws SQLException {
         DbParameterAccessorsMapBuilder params = new DbParameterAccessorsMapBuilder();
 
-        if (objname.contains(".")) {
-            String[] schemaAndName = objname.split("[\\.]", 2);
-            objname = "[" + schemaAndName[0] + "].[" + schemaAndName[1] + "]";
-        } else {
-            objname = "[" + normaliseName(objname) + "]";
+        objname = objname.replaceAll("[^a-zA-Z0-9_.#]", "");
+        String[] qualifiers = objname.split("\\.");
+        String bracketedName = "";
+        for (int i = 0; i < qualifiers.length; i++) {
+            if (i > 0) {
+                bracketedName += ".";
```

Comment:
Ah, you mean something like set it to `"."` initially, then set it to `""` after the first iteration?